src/EventListener/ApiProblemListener.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ApiProblemBundle\EventListener;
+
+use FluentDOM\DOM\Element;
+use Libero\ApiProblemBundle\Exception\ApiProblem;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
+use function strtolower;
+use function substr;
+
+final class ApiProblemListener
+{
+    public function onKernelException(GetResponseForExceptionEvent $event) : void
+    {
+        $request = $event->getRequest();
+        $exception = $event->getException();
+
+        if (!$exception instanceof ApiProblem) {
+            $exception = new ApiProblem($request, $exception);
```

Comment:
Might make sense to move this to a separate listener that runs just before? Could be configured to only convert on a certain path prefix.